### PR TITLE
Refactored TableRecon, Schema, DatabaseConfig Dataclasses to config

### PR DESCRIPTION
**Pull request by vijaypavann-db**
### Changes
------
1. The following classes have been refactored to `config.py`: 
    * TableRecon
    * DatabaseConfig

2. The (import) changes are updated in the corresponding files.

### CODE REVIEW

**Comment by vijaypavann-db** [comment]
Depends on #270

File: src/databricks/labs/remorph/config.py [hunk 1]
```diff
@@ -51,3 +52,29 @@ def get_write_dialect(self):
...
+
+@dataclass
+class DatabaseConfig:
+    source_schema: str
+    target_catalog: str
+    target_schema: str
+    source_catalog: str | None = None
+
+
+@dataclass
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/config.py:77]: I don't think we need schema here for now.

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/config.py (hunk 1 above)
Comment (sundarshankar89) [reply src/databricks/labs/remorph/config.py:77]: also fix fmt errors

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Pull request merged**